src/decoder/mod.rs

```diff
@@ -923,6 +956,38 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(())
     }
 
+    pub fn read_jpeg(&mut self, read_strip_wise: bool) -> TiffResult<DecodingResult> {
```

Comment:
Is the caller ever supposed to know the right value of `read_strip_wise` through some other channel? If not, I'd suggest remove the parameter and moving the code reading the `TileOffset` tag to here.